crates/common/fork_choice/src/store.rs

```diff
@@ -775,8 +775,11 @@ pub fn get_forkchoice_store(
     anchor_state: BeaconState,
     anchor_block: BeaconBlock,
     db: ReamDB,
+    anchor_state_root: Option<B256>,
 ) -> anyhow::Result<Store> {
-    ensure!(anchor_block.state_root == anchor_state.tree_hash_root());
+    let anchor_state_root = anchor_state_root.unwrap_or_else(|| anchor_state.state_root());
+    ensure!(anchor_block.state_root == anchor_state_root);
```

Comment:
```suggestion
    let anchor_state_root = anchor_state.state_root();
    ensure!(anchor_block.state_root == anchor_state_root);
```

I agree it is good to reuse the root calculation, within this function